frontend/ui/uimanager.lua

```diff
@@ -86,6 +86,25 @@ function UIManager:init()
               self.event_handlers["Suspend"]()
             end
         end
+        if not G_reader_settings:readSetting("ignore_power_sleepcover") then
+            self.event_handlers["SleepCoverClosed"] = function()
+                Device.is_cover_closed = true
+                self.event_handlers["Suspend"]()
+            end
+            self.event_handlers["SleepCoverOpened"] = function()
+                Device.is_cover_closed = false
+                self.event_handlers["Resume"]()
+            end
+        else
+            -- Closing/opening the cover will still wake up the device, so we
+            -- need to put it back to sleep if we are in screen saver mode
+            self.event_handlers["SleepCoverClosed"] = function()
```

Comment:
This one is harder to test because when it happens, the device will still show the screensaver screen and have everything locked properly. When the kernel delivers the cover close event, it wakes up the device so koreader can capture it in the main event loop. If the event is simply ignored without calling suspend.sh, koreader will be in a false sleep mode.